source/MaterialXCore/Look.cpp

```diff
@@ -136,4 +158,121 @@ vector<VariantAssignPtr> MaterialAssign::getActiveVariantAssigns() const
     return activeAssigns;
 }
 
+//
+// Lookgroup methods
+//
+
+LookVec LookGroup::getActiveLooks() const
+{
+    string looks = getActiveLook();
+    if (looks.empty())
+    {
+        looks = getLooks();
+    }
+    const StringVec& lookList = splitString(looks, ARRAY_VALID_SEPARATORS);
+    LookVec activeLooks;
+    if (!lookList.empty())
+    {
+        vector<LookPtr> lookElements = getDocument()->getLooks();
+        if (!lookElements.empty())
+        {
+            for (auto lookName : lookList)
+            {
+                for (auto lookElement : lookElements)
+                {
+                    if (lookElement->getName() == lookName)
+                    {
+                        activeLooks.push_back(lookElement);
+                    }
+                }
+            }
+        }
+    }
+    return activeLooks;
+}
+
+void LookGroup::append(const LookGroupPtr& sourceGroup, bool skipDuplicateLooks)
+{
+    string sourceLooks = sourceGroup->getLooks();
+    if (sourceLooks.empty())
+    {
+        return;
+    }
+
+    // If look already exists in the look list then append if
+     // not skipping duplicates
+    const StringVec& sourceLookList = splitString(sourceLooks, ARRAY_VALID_SEPARATORS);
+    StringVec destLookList = splitString(getLooks(), ARRAY_VALID_SEPARATORS);
+    const StringSet destLookSet(destLookList.begin(), destLookList.end());
+
+    for (auto lookName : sourceLookList)
+    {
+        bool appendLook = true;
+        if (destLookSet.count(lookName))
+        {
+            if (skipDuplicateLooks)
+            {
+                appendLook = false;
+            }
+        }
+        if (appendLook)
+        {
+            destLookList.push_back(lookName);
+        }
+    }
+
+    // If look already exists in the active look list then append if not skipping duplicates
```

Comment:
Note I didn't not add in any special casing here to handle the case where we may not want the default look to be here. i.e. it's always additive for now. We should talk about where this particular logic should belong separately.